### PR TITLE
Proposal for a new implementation of  ... [veselin/delta-coefficients]

### PR DESCRIPTION
**Pull request by v-dobrev**
... the support for delta coefficients in linear forms.

Note: this PR updates PR #240.

### CODE REVIEW

File: fem/lininteg.hpp [hunk 1]
```diff
@@ -43,27 +36,61 @@ class LinearFormIntegrator
...
+       multiple elements. */
+   virtual void AssembleDeltaElementVect(const FiniteElement &fe,
+                                         ElementTransformation &Trans,
+                                         Vector &elvect) = 0;
+};
+
+
 /// Class for domain integration L(v) := (f, v)
-class DomainLFIntegrator : public LinearFormIntegrator
+class DomainLFIntegrator : public DeltaLFIntegrator
```
Comment (stefanozampini) [inline fem/lininteg.hpp]: If you're ok with this, I'm fine too

File: fem/lininteg.hpp [hunk 2]
```diff
@@ -153,27 +177,26 @@ class BoundaryTangentialLFIntegrator : public LinearFormIntegrator
 
 /** Class for domain integration of L(v) := (f, v), where
     f=(f1,...,fn) and v=(v1,...,vn). */
-class VectorDomainLFIntegrator : public LinearFormIntegrator
+class VectorDomainLFIntegrator : public DeltaLFIntegrator
```
Comment (stefanozampini) [inline fem/lininteg.hpp]: same as above

**Comment by stefanozampini** [comment]
I briefly tested it today, and it's working perfectly. Better than before I would say. But I did not perform any extensive testing. Give me 24 hours more to do some more experimentations with Nedelec elements.

**Comment by v-dobrev** [comment]
Note: this PR updates PR #240.

**Comment by v-dobrev** [comment]
Thanks to @gelever - he's the one who caught it.

**Pull request merged**